### PR TITLE
Java Lab: update finish button logic

### PR DESCRIPTION
**Pull request by molly-moen**
Overhaul of how we determine if the finish button should be enabled or disabled. Previously the button was enabled unless the page was in code review or was a contained level that hadn't been run yet. Now the logic is:
- On page load the button is always disabled (this matches the behavior of other labs)
- On a level in code review (either your own or a peer's), the finish button is always disabled.
- For non-code reviewing levels the button becomes enabled when all of the following are true
   - The run or test button has been clicked
   - If there is validation code on the level, it has passed (it could have passed on a previous load of the page)

If validation has not passed, the following tooltip shows up on finish button hover:
Screen Shot 2022-07-07 at 11 16 21 AM

This PR will cause eyes diffs because the finish button now starts out disabled.
 
## Links

- jira ticket: [JAVA-322](https://codedotorg.atlassian.net/browse/JAVA-322)

## Testing story
Tested locally and added ui tests.

### CODE REVIEW

File: apps/src/javalab/JavalabView.jsx [hunk 1]
```diff
@@ -208,6 +199,21 @@ class JavalabView extends React.Component {
...
+    // The finish button is disabled if any of the following are true:
+    // 1. The user has not clicked 'run' or test' yet for this session.
+    // 2. This is the user's code and they have opened it for code review.
+    // 4. This is a peer's code, which the current user is code reviewing.
+    // 5. Validation has not passed (if validation does not exist, validationPassed will be true)
+    const disableFinishButton =
+      !hasRunOrTestedCode ||
+      !!hasOpenCodeReview ||
+      !!isCodeReviewing ||
+      !validationPassed;
```
Comment (sanchitmalhotra126) [inline apps/src/javalab/JavalabView.jsx:211]: nit: could move this logic into the `dispatch` function so you don't need to save each of these individually as props? (besides `validationPassed`)

**Review by sanchitmalhotra126 — APPROVED** [review]
LGTM!

File: apps/src/javalab/JavalabView.jsx (hunk 1 above)
Comment (molly-moen) [reply apps/src/javalab/JavalabView.jsx:211]: I think that will make it less readable so I'm going to leave as-is

File: dashboard/config/scripts/levels/Allthethings Java Lab 2.level [hunk 2]
```diff
@@ -11,17 +11,14 @@
         "text": "import java.util.Scanner;\n\npublic class MyClass {\n  public static void main(String[] args) {\n    Scanner myScanner = new Scanner(System.in);\n    System.out.println(\"What's your name?\");\n    String response = myScanner.nextLine();\n    System.out.println(\"Hello \" + response + \"!\");\n  }\n}\n",
         "isVisible": true
       }
-    },
-    "encrypted_examples": [
-
-    ]
+    }
```
Comment (bencodeorg) [inline dashboard/config/scripts/levels/Allthethings Java Lab 2.level:14]: Do you want these changes? Might not matter

File: dashboard/test/ui/features/javalab/code_review_v2_finish_button.feature [hunk 3]
```diff
@@ -0,0 +1,48 @@
+@no_circle
```
Comment (bencodeorg) [inline dashboard/test/ui/features/javalab/code_review_v2_finish_button.feature:1]: Nice tests :)

**Review by bencodeorg — APPROVED** [review]

File: dashboard/config/scripts/levels/Allthethings Java Lab 2.level (hunk 2 above)
Comment (molly-moen) [reply dashboard/config/scripts/levels/Allthethings Java Lab 2.level:14]: I was getting weird errors where it thought this level had validation so I cleaned up the configuration. May have been some conflict in my environment but it shouldn't impact anything to have this cleaned up.

Comment (bencodeorg) [reply dashboard/config/scripts/levels/Allthethings Java Lab 2.level:14]: It looks like only change is removing blank `encrypted_examples` array? Would that tie to validation? Maybe not worth figuring out.

Comment (molly-moen) [reply dashboard/config/scripts/levels/Allthethings Java Lab 2.level:14]: yeah probably not, but it's not needed anyway so I think I'll leave as-is

**Pull request merged**